app/services/folio/locations.rb

```diff
@@ -13,6 +13,53 @@ def self.label(code:)
       instance.data.dig(code, 'discoveryDisplayName')
     end
 
+    # Output a CSV of all locations, flattening objects and converting UUIDs to codes
+    # @return [String] comma-separated data for all locations
+    def self.to_csv
+      headers = %w[institutionCode campusCode libraryCode primaryServicePointCode servicePointCodes]
+
+      data = instance.data.each_value.map do |location|
```

Comment:
Is it weird we're manipulating the original data? Should we change a copy of the location structure instead?

Reply:
Oh, I misunderstood — the name collision on `data` is accidental. I'll change it

Reply:
It depends what else we might do after calling `to_csv`... it just seems like trouble waiting to happen.

Reply:
I'm not sure what you mean about "name collision", but my concern is the totally different data present in ` Folio::Locations.instance.data` before and after calling this method:

```
irb(main):002:0> Folio::Locations.instance.data['GRE-STACKS']
=> 
{"id"=>"4573e824-9273-4f13-972f-cff7bf504217",
 "name"=>"Green Stacks",
 "code"=>"GRE-STACKS",
 "discoveryDisplayName"=>"Stacks",
 "isActive"=>true,
 "institutionId"=>"8d433cdd-4e8f-4dc1-aa24-8a4ddb7dc929",
 "campusId"=>"c365047a-51f2-45ce-8601-e421ca3615c5",
 "libraryId"=>"f6b5519e-88d9-413e-924d-9ed96255f72e",
 "details"=>{},
 "primaryServicePoint"=>"a5dbb3dc-84f8-4eb3-8bfe-c61f74a9e92d",
 "servicePointIds"=>["a5dbb3dc-84f8-4eb3-8bfe-c61f74a9e92d"],
 "servicePoints"=>[],
 "metadata"=>
  {"createdDate"=>"2023-02-08T23:37:29.701+00:00",
   "createdByUserId"=>"7dc1ae38-1e06-4de9-ac86-568b9a2318ec",
   "updatedDate"=>"2023-07-27T09:57:39.746+00:00",
   "updatedByUserId"=>"9c77fffb-9d40-422c-8742-4c0f79957888"}}
irb(main):003:0> Folio::Locations.to_csv; 1
=> 1
irb(main):004:0> Folio::Locations.instance.data['GRE-STACKS']
=> 
{"id"=>"4573e824-9273-4f13-972f-cff7bf504217",
 "name"=>"Green Stacks",
 "code"=>"GRE-STACKS",
 "discoveryDisplayName"=>"Stacks",
 "isActive"=>true,
 "createdDate"=>"2023-02-08T23:37:29.701+00:00",
 "updatedDate"=>"2023-07-27T09:57:39.746+00:00",
 "institutionCode"=>"SU",
 "campusCode"=>"SUL",
 "libraryCode"=>"GREEN",
 "primaryServicePointCode"=>"GREEN-LOAN",
 "servicePointCodes"=>["GREEN-LOAN"]}
```

I don't think we mean to mutate the data, and this kind of thing results in weird and subtle bugs that we should just avoid by working with a copy.

Reply:
Hmm, I see what you mean now. I didn't expect that objects in a `.map` would get passed by reference. I've changed it to just `.dup` first, which seems to fix it.